java/ql/src/Security/CWE/CWE-312/SensitiveStorage.qll

```diff
@@ -243,3 +248,72 @@ class Marshallable extends ClassStore {
     )
   }
 }
+
+/* Holds if the method call is a setter method of `SharedPreferences`. */
+private predicate sharedPreferencesInput(DataFlow::Node sharedPrefs, Expr input) {
+  exists(MethodAccess m |
+    m.getMethod() instanceof SharedPreferencesSetMethod and
+    input = m.getArgument(1) and
+    sharedPrefs.asExpr() = m.getQualifier()
+  )
+}
+
+/* Holds if the method call is the save method of `SharedPreferences`. */
```

Comment:
Store or save?

Reply:
Thanks @smowton for reviewing this PR.

Yes, your understanding is correct. We need to know that this instance of SharedPreferences is specifically not an EncryptedSharedPreferences, which is a subclass that uses the same Editor.

Although there is no existing tests folder <code>java/ql/test/query-tests/security/CWE-312/</code>, I've created one and moved the tests from experimental since the query itself is integrated into existing CWE-312 queries.

And I've changed the text.